test/controllers/EditSetControllerSpec.scala

```diff
@@ -306,6 +306,33 @@ class EditSetControllerSpec extends BaseSpec {
 
       status(editRecordPage) mustBe SEE_OTHER
     }
+
+    "redirect to discard page on form submit even if validation fails" in {
+      val blankScopeAndContentToFailValidation = ""
+      val request = CSRFTokenHelper.addCSRFToken(
+        FakeRequest(POST, "/edit-set/1/record/COAL.2022.V5RJW.P/edit")
```

Comment:
This is out-of-scope, but just an observation: we probably should assert that the record hasn't been changed.

Reply:
Hm, I agree that this kind of test would be useful, however I don't see any spec that tries to model that behavior (or even a spec that asserts behavior across pages).

Maybe we could bring up creating some more specs at the very highest level, e.g.
log in -> edit a set A -> discard -> assert there are no chages on A

Reply:
Yes - as I wrote, it's out-of-scope. Just something to keep in mind going forward.

It can wait for awhile.